kardinal-cli/deployment/deployment.go

```diff
@@ -234,11 +242,43 @@ func DeployKardinalManagerInCluster(ctx context.Context, clusterResourcesURL str
 		return stacktrace.Propagate(err, "An error occurred while applying the kardinal-manager deployment")
 	}
 
+	if err := installGatewayAPI(ctx, kubernetesClientObj); err != nil {
+		return stacktrace.Propagate(err, "An error occurred while installing the Gateway API")
+	}
+
+	return nil
+}
+
+func installGatewayAPI(ctx context.Context, kubernetesClientObj *kubernetes.KubernetesClient) error {
```

Comment:
should we also handle the `gateway API` uninstall?  maybe we can't remove it when the manager is removed but I wonder if we should keep it or not, because it's transparent for users.

Reply:
uninstall for the gateway API would be risk. It could be used by another system in the cluster. While install it should be fine if it was installed before, the uninstalling of Kardinal would also remove it with that other system depending on it.

Reply:
I understand that point.
I wonder if we should clarify somehow for users this, I mean we are handling the installation for them but then they should handle the uninstallation, I mean what happens if a user is testing Kardinal on his cluster, and decides not to continue with it, will it be clear that they will have this new API on the cluster. An alternative could be that they also handle the installation as we do for `Istio and Istio CRDs`

Reply:
ok, I printed information about installing the Gateway and also informed the user if they want to uninstall it, to use a command that also printed

<img width="926" alt="image" src="https://github.com/user-attachments/assets/a08241ee-a099-4b9f-a227-f7700e80cadf">

Reply:
nice!!